src/Ocelot/RateLimiting/MemoryCacheRateLimitStorage.cs

```diff
@@ -1,28 +1,34 @@
 ﻿using Microsoft.Extensions.Caching.Memory;
-
-namespace Ocelot.RateLimit
-{
-    public class MemoryCacheRateLimitCounterHandler : IRateLimitCounterHandler
-    {
-        private readonly IMemoryCache _memoryCache;
-
-        public MemoryCacheRateLimitCounterHandler(IMemoryCache memoryCache)
-        {
-            _memoryCache = memoryCache;
-        }
-
-        public void Set(string id, RateLimitCounter counter, TimeSpan expirationTime)
-        {
-            _memoryCache.Set(id, counter, new MemoryCacheEntryOptions().SetAbsoluteExpiration(expirationTime));
-        }
-
-        public bool Exists(string id) => _memoryCache.TryGetValue(id, out RateLimitCounter counter);
-
-        public RateLimitCounter? Get(string id) => _memoryCache.TryGetValue(id, out RateLimitCounter counter) ? counter : null;
-
-        public void Remove(string id)
-        {
-            _memoryCache.Remove(id);
-        }
-    }
+
+namespace Ocelot.RateLimiting
```

Comment:
Since you refactored the whole rate limiting feature, why not use file scoped namespaces?

Reply:
Indeed, it's important to note that the GitHub diff tool highlights the entire file or namespace block when:
- transitioning between file-scoped and block-scoped namespaces,
- renaming or relocating a code file.

However, the Visual Studio Diff tool, also known as Blame, along with GitHub Blame, function effectively, pinpointing only the precise changes.
This issue is exclusive to GitHub PRs.
Nonetheless, converting to file-scoped namespaces makes sense because of significant refactoring...

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/1592/commits/b862dc310aa9da15eae6f532149cd9d65f6a77dd ✔️ 

#### Files skipped
- `src/Ocelot/RateLimiting/Middleware/RateLimitingMiddleware.cs`: the file is old and I want to point your attention to the exact changes
- all old test classes except new ones